### PR TITLE
Mark imports in __all__ as used

### PR DESCRIPTION
**Pull request by kreathon**
At the moment, symbols that are referenced in string form in the `__all__`  variable assignment are not marked as used.

#### Example

```
# Currently, 'Bar' would be marked as unused.
from foo import Bar

__all__ = ["Bar"]

```

## Implementation

Add `visit_Assign` to cover assignments of the following form:

```
# Default case
__all__ = ["A"]

# Tuples are also tolerated
__all__ = ("B", )
```

#### Limitations

The implementation cannot cover dynamic definitions e.g.:

```
a = ["A"]
__all__ = a
```

## Related Issue
This PR is addressing the issue https://github.com/jendrikseipp/vulture/issues/172.

## Checklist:

- [x] I have updated the documentation in the README.md file or my changes don't require an update.
- [x] I have added an entry in CHANGELOG.md.
- [x] I have added or adapted tests to cover my changes.
- [x] I have run `tox -e fix-style` to format my code and checked the result with `tox -e style`.

### CODE REVIEW

File: vulture/core.py [hunk 1]
```diff
@@ -53,6 +53,16 @@ def _is_test_file(filename):
...
 
 
+def _assigns_special_variable__all__(node):
+    assert isinstance(node, ast.Assign)
+    name_targets = {
+        target.id for target in node.targets if isinstance(target, ast.Name)
+    }
+    return "__all__" in name_targets and isinstance(
+        node.value, (ast.List, ast.Tuple)
+    )
```
Comment (jendrikseipp) [inline vulture/core.py:63]: I can imagine that the following versions runs slightly faster, since it avoids a temporary set and makes the cheap if-test first:

```suggestion
def _assigns_special_variable__all__(node):
    assert isinstance(node, ast.Assign)
    return (
        isinstance(node.value, (ast.List, ast.Tuple)) and
        any(target.id == "__all__" for target in node.targets if isinstance(target, ast.Name))
    )
```

**Review by jendrikseipp — CHANGES_REQUESTED** [review]
Code looks great, thanks! I only have one suggestion, see below. And I was wondering if we should add more tests to cover edge cases such as ignoring code like the following `import my_module; my_module.__all__ = ["Foo"]`. What do you think?

File: vulture/core.py (hunk 1 above)
Comment (kreathon) [reply vulture/core.py:63]: I agree that it runs faster, but in my opinion it is slightly less readable.

But of course I am fine with it ;)

**Comment by kreathon** [comment]
Thank you for your feedback. 

I am  going to add a few more test cases (also to increase the branch coverage).

**Comment by jendrikseipp** [comment]
I agree about the readability, but let's go with that version anyway. Thanks for your great work!
